src/main/java/world/bentobox/bentobox/listeners/flags/protection/TNTListener.java

```diff
@@ -125,6 +111,25 @@ public void onExplosion(final EntityDamageByEntityEvent e) {
         }
     }
 
+    protected boolean protectBlockExplode(Location location) {
+        return getIslands().getProtectedIslandAt(location).map(i -> !i.isAllowed(Flags.BLOCK_EXPLODE_DAMAGE))
+                .orElseGet(() -> !Flags.WORLD_BLOCK_EXPLODE_DAMAGE.isSetForWorld(location.getWorld()));
+    }
+
+    /**
+     * Prevents block explosion from breaking blocks
+     * @param e - event
+     */
+    @EventHandler(priority = EventPriority.LOW, ignoreCancelled = true)
+    public void onExplosion(final BlockExplodeEvent e) {
+        // Check world
+        if (getIWM().inWorld(e.getBlock().getLocation())) {
+            // Remove any blocks from the explosion list if required
+            e.blockList().removeIf(b -> protectBlockExplode(b.getLocation()));
+            e.setCancelled(protectBlockExplode(e.getBlock().getLocation()));
```

Comment:
I think this can be optimized. 

Because if you cancel the event, there is no point to check each block. The event will not be passed and the explosion will be canceled. No point of clearing blocks.

Reply:
If that block was exploded outside the island, that would be different.

Reply:
Because I saw the logic on `EntityExplodeEvent`

Reply:
If you cancel the event, it will not destroy the blocks. So there is no point to check anything in the blocklist. 

If the event is not canceled, you can check the blocks, but not in the opposite way.

Reply:
Well, you also do the block list first and check if any block is left in the list.
If the list is empty, cancel the event.

It would allow exploding blocks outside island boundaries. 

Because currently, explosion will be stopped if `e.getBlock().getLocation()` returns as protected.

Reply:
That should be it.